lib/ravelin/event.rb

```diff
@@ -0,0 +1,119 @@
+module Ravelin
+  class Event
+    attr_accessor :name, :timestamp, :payload
+
+    def initialize(name:, payload:, timestamp: nil)
+      @name = name.to_sym
+      @payload = convert_to_ravelin_objects(payload)
+      @timestamp = timestamp.nil? ? Time.now.to_i : convert_to_epoch(timestamp)
+
+      validate_top_level_payload_params
+    end
+
+    def serialize
```

Comment:
`def to_hash` instead? The use of this in the `Ravelin` initializer is a little confusing as (to me) serialisation implies converting to a string or bytes and there you're then converting to json.